topics/tutorials/code_documentation.md

```diff
@@ -0,0 +1,33 @@
+[//]: # (title: Documentation)
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+[Quick Documentation](https://www.jetbrains.com/help/idea/viewing-reference-information.html#inline-quick-documentation)
+helps users by showing documentation, e.g., for classes, functions, or methods inside the editor.
+Plugin authors implement
+[`DocumentationProvider`](upsource:///platform/analysis-api/src/com/intellij/lang/documentation/DocumentationProvider.java)
+to show documentation for particular Psi elements.
+
+`DocumentationProvider` can be used for existing languages as well as for a custom language implementation.
+For an existing language, the implementation of `DocumentationProvider` is registered as `com.intellij.documentationProvider` extension point (EP),
+while custom languages should register it as `com.intellij.lang.documentationProvider` EP.
+The difference between both is that the latter requires the `language` attribute when registering the EP in the `plugin.xml` and is only called
+for this particular language.
+
+Several documentation providers can co-exist for the same language, and plugin authors can implement additional ones even for languages not under
+their control.
+If there is more than one provider for the same element, then the first one that returns a value different from `null` wins.
+Although discouraged, the ordering of documentation providers can be influenced by using the `order` attribute when registering the extension.
+For instance, [`python-core-common.xml`](upsource:///python/src/META-INF/python-core-common.xml) uses the following to call the external documentation
+provider before the default one:
+
+```xml
+<lang.documentationProvider
```

Comment:
I like this formatting style (except closing token in a new line) and use it personally, but currently, we align attributes to the first one which is placed in the same line as the element name. I would keep it consistent and would propose changing XML snippets formatting in all docs as this style is more likely to fit in the code snippets widget.

It's minor though, feel free to ignore it.

Reply:
I'm not sure how important it is to be consistent throughout all the docs, but for cases like this one, I really like that style. 2 spaces of indent to make as much of the code visible without scrolling.